### PR TITLE
Add publishing build artifacts to azure blob storage

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
This PR adds new step with publishing of build artifacts into BLOB storage.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -57,6 +62,18 @@ jobs:
...
+    inputs:
+      targetType: filePath
+      filePath: "./bin/ci/az-publish.sh"
+    env:
+      AZURE_STORAGE_ACCOUNT: $(AzureStorageAccount)
+      AZURE_STORAGE_KEY: $(AzureStorageKey)
+      AZURE_STORAGE_CONNECTION_STRING: $(AzureStorageConnectionString)
+      SOURCE_BRANCH: $(Build.SourceBranch)
+    displayName: "Publish to Azure Blob Storage"
+    condition: eq(variables['Agent.JobName'], variables['AzurePublishWhen'])
```
Comment (ivan-nosar) [inline azure-pipelines.yml]: Please use the following condition: `condition: and(succeeded(), eq(variables['Agent.JobName'], variables['AzurePublishWhen']), startsWith(variables['Build.SourceBranch'], 'master'))`

File: bin/ci/az-publish.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+source bin/log.sh
```
Comment (ivan-nosar) [inline bin/ci/az-publish.sh]: I believe this is not necessary

File: bin/ci/az-publish.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+source bin/ditto.sh
+source bin/simctl.sh
```
Comment (ivan-nosar) [inline bin/ci/az-publish.sh]: This is unnecessary too

File: bin/simctl.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (ivan-nosar) [inline bin/simctl.sh]: Why does we need this file here?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -57,6 +62,18 @@ jobs:
...
+    inputs:
+      targetType: filePath
+      filePath: "./bin/ci/az-publish.sh"
+    env:
+      AZURE_STORAGE_ACCOUNT: $(AzureStorageAccount)
+      AZURE_STORAGE_KEY: $(AzureStorageKey)
+      AZURE_STORAGE_CONNECTION_STRING: $(AzureStorageConnectionString)
+      SOURCE_BRANCH: $(Build.SourceBranch)
+    displayName: "Publish to Azure Blob Storage"
+    condition: and(succeeded(), eq(variables['Agent.JobName'], variables['AzurePublishWhen']), startsWith(variables['Build.SourceBranch'], 'master'))
```
Comment (jmoody) [inline azure-pipelines.yml]: As discussed, we want to add `IndividualCI` to this condition.

Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

**Pull request merged**